src/safeposix/syscalls/fs_calls.rs

```diff
@@ -389,42 +389,121 @@ impl Cage {
+        // the metadata of a given inode from the Inode Table.
         let metadata = &FS_METADATA;
 
+        // Walk through the absolute path which returns a tuple consisting of inode
+        // number of file (if it exists), and inode number of parent (if it exists)
         match metawalkandparent(truepath.as_path()) {
-            //If neither the file nor parent exists
-            (None, None) => syscall_error(
-                Errno::ENOENT,
-                "mknod",
-                "a directory component in pathname does not exist or is a dangling symbolic link",
-            ),
-
-            //If the file doesn't exist but the parent does
+            // Case: When the file doesn't exist but the parent directory exists
             (None, Some(pardirinode)) => {
-                let filename = truepath.file_name().unwrap().to_str().unwrap().to_string(); //for now we assume this is sane, but maybe this should be checked later
-
-                //assert sane mode bits (asserting that the mode bits make sense)
+                // for now we assume this is sane, but maybe this should be checked later
+                let filename = truepath.file_name().unwrap().to_str().unwrap().to_string();
+
+                // S_FILETYPEFLAGS represents a bitmask that can be used to extract
+                // the file type information from a file's mode.
+                // This code is referenced from Lind-Repy codebase.
+                // Here, we are checking whether the mode bits are sane by ensuring
+                // that only valid file permission bits (S_IRWXA) and file type bits
+                // (S_FILETYPEFLAGS) are set. Else, we return the error.
                 if mode & (S_IRWXA | S_FILETYPEFLAGS as u32) != mode {
-                    return syscall_error(Errno::EPERM, "mknod", "Mode bits were not sane");
+                    return syscall_error(
+                        Errno::EPERM, 
+                        "mknod", 
+                        "Mode bits were not sane"
+                    );
                 }
+
+                // As of now, the only file type in LIND supported by mknod_syscall
+                // is "Char Device" represented by S_IFCHR flag.
+                // In order to check for Char file type, a bitwise-AND operation for
+                // S_IFCHR flag is performed with the "mode" bits and an error is returned
+                // when the result is 0 denoting the support for only character files.
                 if mode as i32 & S_IFCHR == 0 {
                     return syscall_error(
                         Errno::EINVAL,
                         "mknod",
                         "only character files are supported",
                     );
                 }
-                let time = interface::timestamp(); //We do a real timestamp now
+                // New Inode of type CharDev is created with file size 0
+                let time = interface::timestamp(); // We do a real timestamp now
```

Comment:
Why use the real time?  (I'm not complaining, just curious)

Reply:
Some programs in our code require logical times for execution, so we use real time for it. Also, Nginx has an assert condition, which requires time to be moving forward. 
Alternatively, we can use `clock counter` for it.